dra-evolution/pkg/api/capacity_types.go

```diff
@@ -1,170 +1,205 @@
 package api
 
 import (
+	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/types"
 )
 
 const (
 	DevMgmtAPIVersion = "devmgmtproto.k8s.io/v1alpha1"
 )
 
-// DevicePool represents a collection of devices managed by a given driver. How
+// ResourcePool represents a collection of devices managed by a given driver. How
 // devices are divided into pools is driver-specific, but typically the
 // expectation would a be a pool per identical collection of devices, per node.
 // It is fine to have more than one pool for a given node, for the same driver.
-type DevicePool struct {
-	metav1.TypeMeta   `json:",inline"`
-	metav1.ObjectMeta `json:"metadata,omitempty"`
-
-	Spec   DevicePoolSpec   `json:"spec,omitempty"`
-	Status DevicePoolStatus `json:"status,omitempty"`
-}
-
-// DevicePoolSpec identifies the driver and contains the data for the pool
-// prior to any allocations.
-// NOTE: It's not clear that spec/status is the right model for this data.
-type DevicePoolSpec struct {
-	// NodeName is the name of the node containing the devices in the pool.
-	// For network attached devices, this may be empty.
+type ResourcePool struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object metadata
 	// +optional
-	NodeName *string `json:"nodeName,omitempty"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
 
-	// Driver is the name of the DeviceDriver that created this object and
-	// owns the data in it.
-	// +required
-	Driver string `json:"driver,omitempty"`
-
-	// Attributes contains device attributes that are common to all devices
-	// in the pool.
-	// +optional
-	// +listType=atomic
-	Attributes []Attribute `json:"attributes,omitempty"`
-
-	// SharedResources are pooled resources that are shared by all devices
-	// in the pool. This is typically used when representing a
-	// partitionable device, and need not be populated otherwise.
+	// NodeName identifies the node which provides the devices
+	// if (and only if) they are local to a node. If this is empty,
+	// the devices exist outside of any specific node.
+	//
+	// NodeName and NodeSelector are mutually exclusive. If none is
```

Comment:
I disagree here, for two reasons:
- Akri has been an early supporter of the ideas behind DRA because they needed it to model network-attached IP cameras. Telling them "you don't matter, we do GPUs now" would be pretty harsh.
- It's not that hard to support, as long as we limit it to "one scheduler per cluster" (which is my proposal). We just need to have the `NodeSelector` as additional field and then some logic around which nodes the pool can be used for.

Reply:
See [my proposal](https://github.com/kubernetes-sigs/wg-device-management/pull/14#discussion_r1615713520) to define the extension point for `nodeSelector` in a one-of without implementing support right now.